Sources/SwiftDocCUtilities/ArgumentParsing/Subcommands/Merge.swift

```diff
@@ -0,0 +1,131 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import ArgumentParser
+@_spi(FileManagerProtocol) import SwiftDocC
+import Foundation
+
+extension Docc {
+    /// Merge a list of documentation archives into a combined archive.
+    public struct Merge: ParsableCommand {
+        public init() {}
+        
+        public static var configuration = CommandConfiguration(
+            abstract: "Merge a list of documentation archives into a combined archive."
+        )
+        
+        private static let archivePathExtension = "doccarchive"
+        private static let catalogPathExtension = "docc"
+        
+        // The file manager used to validate the input and output directories.
+        //
+        // Provided as a static variable to allow for using a different file manager in unit tests.
+        static var _fileManager: FileManagerProtocol = FileManager.default
+        
+        // Note:
+        // The order of the option groups in this file is reflected in the 'docc merge --help' output.
+        
+        // MARK: - Inputs & outputs
+        
+        @OptionGroup(title: "Inputs & outputs")
+        var inputsAndOutputs: InputAndOutputOptions
+        struct InputAndOutputOptions: ParsableArguments {
+            @Argument(
+                help: ArgumentHelp(
+                    "A list of paths to '.\(Merge.archivePathExtension)' documentation archive directories to combine into a combined archive.",
+                    valueName: "archive-path"),
+                transform: URL.init(fileURLWithPath:))
+            var archives: [URL]
+            
+            @Option(
+                help: ArgumentHelp(
+                    "Path to a '.\(Merge.catalogPathExtension)' documentation catalog directory with content for the landing page.",
+                    valueName: "catalog-path"),
+                transform: URL.init(fileURLWithPath:))
+            var landingPageCatalog: URL?
+            
+            @Option(
+                name: [.customLong("output-path"), .customShort("o")],
+                help: "The location where the documentation compiler writes the combined documentation archive.",
+                transform: URL.init(fileURLWithPath:)
+            )
+            var providedOutputURL: URL?
+            
+            var outputURL: URL!
+            
+            mutating func validate() throws {
+                let fileManager = Docc.Merge._fileManager
+                
+                guard !archives.isEmpty else {
```

Comment:
Does it make sense to merge just one archive? Shouldn't we also raise a validation error when `archives.count == 1` ?

Reply:
Yes, merging a single archive is supported. Doing so will inactivate all the external links in that archive.